core/io/close_spec.rb

```diff
@@ -45,6 +45,26 @@
       @io.close.should be_nil
     end
   end
+
+  ruby_version_is '2.5.0' do
+    it 'raises an IOError when the stream is closed in another thread' do
+      read_io, write_io = IO.pipe
+      thread = Thread.new do
+        begin
+          read_io.read
+        rescue => e
+          e
+        end
+      end
+
+      Thread.pass until thread.stop?
+      read_io.close
+      thread.value.message.should == 'stream closed in another thread'
+      thread.value.class == IOError
```

Comment:
Could you move those checks in the Thread.new block and use `should raise_error(IOError, 'stream closed in another thread')` ?
That would simplify the logic a bit I think.

Reply:
done. I have updated the PR.